framework/base/View.php

```diff
@@ -332,13 +332,26 @@ public function renderPhpFile($_file_, $_params_ = [])
      * Renders dynamic content returned by the given PHP statements.
      * This method is mainly used together with content caching (fragment caching and page caching)
      * when some portions of the content (called *dynamic content*) should not be cached.
-     * The dynamic content must be returned by some PHP statements.
+     * The dynamic content must be returned by some PHP statements. You can optionally pass
+     * additional parameters that will be available as variables in the PHP statement:
+     *
+     * ```php
+     * <?= $this->renderDynamic('return foo($myVar);', [
+     *     'myVar' => $model->getMyComplexVar(),
+     * ]) ?>
+     * ```
      * @param string $statements the PHP statements for generating the dynamic content.
+     * @param array $params the parameters (name-value pairs) that will be extracted and made
+     * available in the $statement context. The parameters will be stored in the cache and be reused
+     * each time $statement is executed. You should make sure, that these are safely serializable.
      * @return string the placeholder of the dynamic content, or the dynamic content if there is no
      * active content cache currently.
      */
-    public function renderDynamic($statements)
+    public function renderDynamic($statements, $params = null)
```

Comment:
I've made a test for `empty($array)` and `$array !== []` for both empty and non-empty arrays at https://3v4l.org/.